com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -48,15 +50,30 @@ internal struct AgentInfo
         /// to separate between different agents in the environment.
         /// </summary>
         public int episodeId;
-    }
 
-    /// <summary>
-    /// Struct that contains the action information sent from the Brain to the
-    /// Agent.
-    /// </summary>
-    internal struct AgentAction
-    {
-        public float[] vectorActions;
+        public void ClearActions()
+        {
+            Array.Clear(storedVectorActions, 0, storedVectorActions.Length);
+        }
+
+        public void CopyActions(float[] continuousActions, int[] discreteActions)
+        {
+            var start = 0;
+            if (continuousActions != null)
+            {
+                Array.Copy(continuousActions, 0, storedVectorActions, start, continuousActions.Length);
+                start = continuousActions.Length;
+            }
+            if (start >= storedVectorActions.Length)
+            {
+                return;
+            }
+
+            if (discreteActions != null)
+            {
+                Array.Copy(discreteActions, 0, storedVectorActions, start, discreteActions.Length);
```

Comment:
How does the copy work here for mixed types? Is it a memcpy, or does it actually do the int->float conversion?

Reply:
Never mind, [found it](https://docs.microsoft.com/en-us/dotnet/api/system.array.copy?view=netcore-3.1).
> performs type casting and boxing as required